File: university-work/before/liceu/clsx/div&imp/caut_bin.pas
Program cautare_binara;
uses crt;
var x:array[1..100] of integer;
    n,a,i:integer;
    q:byte;

Function cauta(s,d:byte):byte;
var r:byte;
begin
 cauta:=0;
 while s<=d do
  begin
   if a=x[(s+d) div 2] then
                        begin
                         cauta:=(s+d) div 2;
                         s:=5;d:=4;
                        end
                       else if a < x[(s+d) div 2] then d:=(s+d) div 2 - 1
                                                  else s:=(s+d) div 2 + 1;
  end;
end;


begin
 clrscr;
 writeln;
 write('Numarul de elemente din sir : ');readln(n);
 writeln('Introduceti sirul.Atentie, trebuie sa fie ordonat crescator.');
 for i:=1 to n do
  begin
   write('Elementul ',i,' : ');
   readln(x[i]);
  end;
 write('Numarul ce trebuie cautat in sir : ');readln(a);
 q:=cauta(1,n);
 if q<>0 then writeln('Numarul este pe pozitia ',q,'.')
         else writeln('Numarul nu se afla in sir.');
 readkey;
end.

File: university-work/before/c/div&imp/caut_bin.cpp
#include <stdio.h>
#include <conio.h>

cautare_binara(int n, int a[100], float x)
{
   int p, u, i;

   p= 1;
   u= n;
   while (p <= u)
   {
      i= int((p+u)/2);
      if (x == a[i])
	 return i;
      else
	 if (x < a[i])
	    u= i-1;
	 else
	    p= i+1;
   }

  return 0;
}

main()
{
   int s, i, n;
   int a[100];
   float x;

   clrscr();
   printf("\n");

   printf(" Numarul de elemente : ");
   scanf("%i", &s);
   for (i= 1; i <= s; i= i+1)
   {
      printf("  Elementul %i : ", i);
      scanf("%i", &a[i]);
   }
   printf(" Elementul ce trebuie cautat : ");
   scanf("%f", &x);

   n= cautare_binara(s, a, x);
   if (n != 0)
      printf(" Elementul exista in sir pe pozitia %i.", n);
   else
      printf(" Elementul nu exista in sir.");

   getch();

   return 0;
}